src/module.c

```diff
@@ -10693,7 +10693,8 @@ size_t RM_MallocSize(void* ptr) {
 /* Similar to RM_MallocSize, the difference is that RM_MallocUsableSize
  * returns the usable size of memory by the module. */
 size_t RM_MallocUsableSize(void *ptr) {
-    return zmalloc_usable_size(ptr);
+    /* todo: We cannot protect the module user from writing additional memory. */
+    return zmalloc_usable_size_safe(&ptr);
```

Comment:
The safe thing to do here might be to also update `RM_Alloc`, `RM_Calloc`, etc. to do an `extend_to_usable` so that only the usable size is seen at all times.

However, AFAICT the RedisModule functions don't show themselves as allocators (i.e. no `alloc_size` attribute) so they're unlikely to trip over this.

Reply:
@sundb IMHO this is what we should do:
1. eliminate zmalloc_usable_size_safe (it's odd, and doesn't really help here).
2. add the malloc_size attribute to RM_Alloc and RM_Realloc, RM_TryAlloc, RM_Calloc
3. since we can't change the semantics of RM_MallocUsableSize and document how it should be used (i.e. together with extend_to_usable, what we should do is make sure RM_Alloc family uses extend_to_usable or better yet zmalloc_usable internally before returning the pointer to the module (and add some comment explaining it)